### PR TITLE
Fix travis build / Update Buffer usage

### PR DESCRIPTION
**Pull request by holgerd77**
This fixes the travis build by updating the Node versions tested and use non-deprecated (safe-buffer) initialization.

### CODE REVIEW

**Comment by fanatid** [comment]
utACK

File: provider-engine.js [hunk 1]
```diff
@@ -14,10 +14,10 @@ function WalletSubprovider (wallet, opts) {
 
   opts.getPrivateKey = function (address, cb) {
     if (address !== wallet.getAddressString()) {
-      return cb('Account not found')
+      cb(new Error('Account not found'))
+    } else {
+      cb(null, wallet.getPrivateKey())
     }
-
-    cb(null, wallet.getPrivateKey())
```
Comment (axic) [inline provider-engine.js]: This seems to be a non-relevant change (to the comment message).

File: thirdparty.js [hunk 2]
```diff
@@ -28,7 +29,7 @@ var Thirdparty = {}
  *
  * FIXME: not optimised at all
  */
-function evp_kdf (data, salt, opts) {
+function evpKdf (data, salt, opts) {
```
Comment (axic) [inline thirdparty.js]: This seems to be an irrelevant change.

**Comment by axic** [comment]
I'm not sure I'd like to remove support for older Node versions.

File: thirdparty.js (hunk 2 above)
Comment (holgerd77) [reply thirdparty.js]: This fixes linting.

Comment (axic) [reply thirdparty.js]: Since when does it have a rule of not using underscores? Seems very weird.

**Comment by holgerd77** [comment]
There's only Node 4 which would still be in the supported versions range, will re-add that and see if the tests still pass.

**Comment by holgerd77** [comment]
@axic Can't do, have no write access on this any more.

File: thirdparty.js (hunk 2 above)
Comment (holgerd77) [reply thirdparty.js]: Names have to be ``camelCase``. My favorite rule for lint-hating is actually this trailing-spaces rule. :-)

**Comment by holgerd77** [comment]
@axic What's the status of this now?

**Comment by holgerd77** [comment]
@axic ?

File: thirdparty.js (hunk 2 above)
Comment (holgerd77) [reply thirdparty.js]: @axic I've now un-done the name change from above to avoid side effects (unlikely though). This should now ready to be merged.

**Comment by holgerd77** [comment]
On Node 4 support: I think we should have found a broad consensus over the last weeks/2-3 month that Node 4 will not be supported any more.
